check/README.md

```diff
@@ -4,20 +4,23 @@
 
 - `string table` a file with translation
 - `reference table` en.txt string table.
-  English is the main game language, so any changes related to text should be in that file.
-  Also this table us used as fallback for other translation.
+  English is the development language, so any changes related to text should be in that file.
+  Also, this table us used as fallback for other translations, if a needed key is missing.
   This file is mostly maintained by people who change game code,
   so the list of entries in this file should be the most actual.
 - `entry` one piece of translation, it consists of
   - `key` a sting that represents entry in the codebase (C++, Python).
     This string must be unique.
   - `value` actual translation that will be shown to the user.
 
-
 ### Key
 Only English chars in uppercase and dash are allowed.
-Rename of a key should be done with caution. Usually key used as is in code,
-but sometimes it could be used as prefix and code logic based on name could be present.
+Renaming a key should be done with caution.
+Many of the keys in the stringtable are used in the game source code explicitly,
+or in content scripts.
+Some keys are also generated programatically, such as by adding a prefix to other text.
+C++ use `UserString` and `UserStringNop` to retrieve keys,
```

Comment:
C++ uses `UserString(...)` or `UserStringNop(...)` to look up strings or to mark stringtable keys that are used in code.

Reply:
@geoffthemedio, i am afraid it is not used consequently enough. there are still thousands of places where we need to put those in, in order for e.g. the japanese translation (which someone is working on right now) to look and feel japanese-ish, all the punctuation and spaces and things that are just programmatically done to and with strings ...

Reply:
maybe can write a test script to check that each and every single string literal is marked as either `UserString(...)` or `UserStringNop(...)` but not let the build fail, just to get a list of warnings

Reply:
> in order for e.g. the japanese translation (which someone is working on right now) to look and feel japanese-ish, all the punctuation and spaces and things that are just programmatically done to and with strings ...

These are mostly unrelated issues, aren't they? Yes, it's impossible to do a proper looking translation by replacing substrings of english text with another language when there are incompatible built-in assumptions about grammar and formatting and such in the code. But how many of the strings that are used in the C++ code have `UserStringNop` wrappers is mostly independent of that.

Reply:
> maybe can write a test script to check that each and every single string literal is marked as either `UserString(...)` or `UserStringNop(...)`

Not all string literals in the code are expected to have translations. The point is to flag or mark the ones that should, when they appear in the C++ code explicitly and exactly as there should be corresponding keys found in the stringtable.

There are actually numerous incorrect uses, like [here](https://github.com/freeorion/freeorion/blob/master/client/AI/AIClientApp.cpp#L54), where the key is not written explicitly in code but is being programatically generated. In those cases, wrapping with `UserStringNop` isn't necessary or useful, as there's no trivial way to determine from that C++ code what specific strings should be present.

It would probably be useful to have a test script that checks through all the C++ code for appearances of `UserString("...")` and `UserStringNop("...")` and makes a list, and then checks if the corresponding strings are present in the stringtables. That was the motivation for adding the `Nop` version of the wrapper, at least as far as I recall.